### DISCUSSION TITLE
Writing tests

### DISCUSSION THREAD

**Discussion by 0x-codeX** [Cyfrin/foundry-full-course-cu#1598] [Q&A]
I don't know if this is just me. I am almost through with Lesson 12 meaning I have written quite several tests throughout the course but I do not seem to get a hand with a system or process of how tests are written, I pretty much understand a lot of other stuff I can even do most myself except writing tests. I guess what I am asking is how do I know how to write a test for a particular function, as there have been many several ways Patrick has written tests all over the course, some with VM and some without, have I missed a lecture or something, or is it just me??

**Comment by EngrPips** [comment]
Hello @codexNature , The only way to get good at anything is to do it over and over and over again. repetition is the mother of skill. Most importantly nobody can teach you how to be good at anything as everyone have their own way of doing things , so do it your way and do it well enough that you are good at it.

**Reply by 0x-codeX** [reply]
Thank you @EngrPips I do agree that repetition is the mother of skill and I can say that I have most of Solidity programming down as I have committed myself to code daily for over two months and I always have things jump out to me and I completely understand why and how things work and I have that down with most of Solidity, it is just different with the test, I think I have only been able to write two test myself before Patricks writes them successfully. Well like you said I will keep keeping on, I'm sure I'll eventually have it locked down. Thanks again.

**Reply by EngrPips** [reply]
Exactly, You would get it locked down if you keep at it.
